src/main/plugin/iso19139.ca.HNAP/layout/utility-tpl-multilingual.xsl

```diff
@@ -47,7 +47,15 @@
             <xsl:value-of select="$metadata/gmd:language/gmd:LanguageCode/@codeListValue"/>
           </xsl:when>
           <xsl:when test="contains($metadata/gmd:language/gco:CharacterString,';')">
-            <xsl:value-of  select="normalize-space(substring-before($metadata/gmd:language/gco:CharacterString,';'))"/>
+            <xsl:variable name="metadataMainLanguage" select="normalize-space(substring-before($metadata/gmd:language/gco:CharacterString,';'))"/>
```

Comment:
There is a function for getting the main language and it does more than just taking the language before the ; - it would be better if you used that function if possible.

Reply:
I was following what's been done and attempt to make minimum change. Which function are you referring to by the way?

Reply:
I'm hoping one of these would work.

Here are 2 from HNAP

https://github.com/metadata101/iso19139.ca.HNAP/blob/564a276ecd1d6d53cef7aad640ce2ce84bfa7056/src/main/plugin/iso19139.ca.HNAP/convert/functions.xsl#L39-L55

https://github.com/metadata101/iso19139.ca.HNAP/blob/564a276ecd1d6d53cef7aad640ce2ce84bfa7056/src/main/plugin/iso19139.ca.HNAP/layout/utility-tpl-multilingual.xsl#L38-L58

Here and similar from iso19139

https://github.com/geonetwork/core-geonetwork/blob/ce3d7d5f88aeb58404b695c8c4ed8142a78a1e4c/schemas/iso19139/src/main/plugin/iso19139/convert/functions.xsl#L149-L165

https://github.com/geonetwork/core-geonetwork/blob/ce3d7d5f88aeb58404b695c8c4ed8142a78a1e4c/schemas/iso19139/src/main/plugin/iso19139/layout/utility-tpl-multilingual.xsl#L35-L55

Reply:
Confimed, it works. I will commit the code for it.